lib/services/remove_watcher.rb

```diff
@@ -36,6 +36,11 @@ def run(success: -> {}, failure: -> {})
     if @work_package.watcher_users.include?(@user)
       @work_package.watcher_users.delete(@user)
```

Comment:
If the watcher for the user where to be retrieved first, and then destroyed, that destroyed object could then be passed on through the notification without the need to bloat up the interface from three parameters `watcher_toggled`, `watcher`, `watcher_setter` to five. 
The same would then be true for the `WatcherNotificationMailer.handle_watcher_toggle` method.